File: temp.py
class Problem:
    def __init__(self, n, m):
        self.n = n
        self.m = m
        self.answer = [0 for _ in range(m)]
        self.used = [False for _ in range(n)]

    def printAnswer(self):
        for number in self.answer:
            print(number,end=' ')
        print()

    def makeAnswer(self, index):
        if index == len(self.answer):
            self.printAnswer()
            return
        for i in range(1, n+1):
            if index == 0 or (self.used[i-1] == False and self.answer[index-1]<i):
                self.answer[index] = i
                self.used[i-1] = True
                self.makeAnswer(index+1)
                self.used[i-1] = False
                self.answer[index] = 0
        return

    def printAnswers(self):
        self.makeAnswer(0)
        return


n, m = map(int, input().split())
problem = Problem(n, m)
problem.printAnswers()

File: temp.js
class Problem{
    constructor(n, m){
        this.n = Number(n);
        this.m = Number(m);
        this.answer = new Array(m);
        for(let i = 0;i<this.answer.length;i++){
            this.answer[i] = 0;
        }
        this.used = new Array(n);
        for(let i = 0;i<this.used.length;i++){
            this.used[i] = false;
        }
    }

    printAnswer(){
        let answerNum = ''
        for(let i = 0;i<this.answer.length;i++){
            answerNum+=String(this.answer[i])
            answerNum+=' '
        }
        console.log(answerNum);
    }

    makeAnswer(index){
        if(index == this.answer.length){
            this.printAnswer();
            return;
        }

        for(let i=1;i<=this.n;i++){
            if(
                index == 0 || 
                (
                    !this.used[i-1] && 
                    this.answer[index - 1] < i
                )
            ){
                this.answer[index] = i;
                this.used[i-1] = true;
                this.makeAnswer(index+1);
                this.used[i-1] = false;
                this.answer[index] = 0;
            }
        }
        return;
    }

    printAnswers(){
        this.makeAnswer(0);
        return;
    }
}

const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let n, m;

rl.on("line", line=>{
    const inputs = line.split(' ');
    n = Number(inputs[0]);
    m = Number(inputs[1]);
}).on("close", ()=>{
    const problem = new Problem(n, m);
    problem.printAnswers();
})

